### PR TITLE
E2E: Make sure element presence helpers are used correctly

### PR DESCRIPTION
**Pull request by WunderBart**
#### Changes proposed in this Pull Request

Make sure element presence helpers are used correctly so no false-negative result is produced.

`isElement*` helpers only return a boolean value, they won't throw in any case. To verify an element is or isn't present a `waitUntilElement*` helpers should be used.

#### Testing instructions

All tests should pass.

### CODE REVIEW

**Review by worldomonation — APPROVED** [review]

File: test/e2e/lib/pages/signup/signup-processing-page.js [hunk 1]
```diff
@@ -17,18 +17,6 @@ export default class SignupProcessingPage extends AsyncBaseContainer {
 		super( driver, By.css( '.signup-processing-screen__content' ) );
 	}
 
-	async waitForPage() {
```
Comment (scinos) [inline test/e2e/lib/pages/signup/signup-processing-page.js:20]: Just to confirm: this was deleted intentionally because nobody is using, right?

**Review by scinos — APPROVED** [review]

**Comment by fullofcaffeine** [comment]
🚢 

(A few specs files were [deleted](https://github.com/Automattic/wp-calypso/pull/52895) from trunk and it caused a conflict in the base branch for this PR. I rebased this branch from the base branch (which had got a rebase from trunk, see: https://github.com/Automattic/wp-calypso/pull/53048).

**Review by fullofcaffeine — APPROVED** [review]
🚢 

(A few specs files were [deleted](https://github.com/Automattic/wp-calypso/pull/52895) from trunk and it caused a conflict in the base branch for this PR. I rebased this branch from the base branch (which had got a rebase from trunk, see: https://github.com/Automattic/wp-calypso/pull/53048)).

File: test/e2e/lib/pages/signup/signup-processing-page.js (hunk 1 above)
Comment (fullofcaffeine) [reply test/e2e/lib/pages/signup/signup-processing-page.js:20]: Searching for `signupProcessingPage` and `SignupProcessPage` shows this method is indeed not being used anymore.

```
test/e2e/lib/flows/sign-up-flow.js
42:             const signupProcessingPage = await SignupProcessingPage.Expect( this.driver );
43:             await signupProcessingPage.waitToDisappear( this.accountName, this.password );

test/e2e/lib/flows/sign-up-step.js
21:             const signupProcessingPage = await SignupProcessingPage.Expect( this.driver );
22:             return await signupProcessingPage.waitToDisappear( blogName, passwordForTestAccounts );
```

**Review by scinos — APPROVED** [review]

**Pull request merged**